src/SelectMenu/SelectMenu.js

```diff
@@ -0,0 +1,71 @@
+import React, {useRef, useState} from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import {COMMON} from '../constants'
+import theme from '../theme'
+import {wrapperStyles} from './SelectMenuStyles'
+import {MenuContext} from './SelectMenuContext'
+import SelectMenuDivider from './SelectMenuDivider'
+import SelectMenuFilter from './SelectMenuFilter'
+import SelectMenuFooter from './SelectMenuFooter'
+import SelectMenuItem from './SelectMenuItem'
+import SelectMenuList from './SelectMenuList'
+import SelectMenuModal from './SelectMenuModal'
+import SelectMenuTabs from './SelectMenuTabs'
+import SelectMenuTab from './SelectMenuTab'
+import SelectMenuTabPanel from './SelectMenuTabPanel'
+import useKeyboardNav from './hooks/KeyboardHook'
+
+const SelectMenuBase = ({children, initialTab, theme, ...rest}) => {
+  const ref = useRef(null)
+  const [selectedTab, setSelectedTab] = useState(initialTab)
+  const [open, setOpen] = useState(false)
+  useKeyboardNav(ref)
+  const menuProviderValues = {
+    selectedTab,
+    setSelectedTab,
+    setOpen,
+    open,
+    initialTab
+  }
+
```

Comment:
Since this a non-controlled component, I think a blanket `onTabChanged` prop could be useful on `SelectMenu`. This could be added to the context and called in `SelectMenu.Tab`, similar to its `onClick` handler — but not sure if it makes sense to have both of these. Don't feel strongly about it but I think my gut is that a single `onTabChanged` callback is a little more ergonomic than individual `onClick` handlers.

Reply:
Curious what @clayplumridge thinks - my initial thought on having separate onClick handlers was to allow users to trigger different events depending on which tab was clicked but not sure if that would be more ergonomic to use the `selectedTab` and `onTabChanged` state instead.

Reply:
Why not both?
In the tab itself:
```
const onClickWrapped = (event: React.MouseEvent<HTML???Element>) => {
  props.onClick && props.onClick(event)

  if(!event.defaultPrevented) {
    context.onTabChanged(event, props.id)
  }
}
```

and in the wrapping component:
```
const onTabChanged = (event: React.MouseEvent<HTML???Element>, newTabId: string) => {
  props.onTabChanged && props.onTabChanged(event, newTabId)
  if(!event.defaultPrevented) {
    setTabSelected(newTabId)
  }
}
```

I am GENERICALLY in favor of having more controlled components, which in this case would mean passing the selected tab in, but this strikes me as the best middle-ground if you want this to be a non-controlled component.

Reply:
(I can see use cases where it's more obvious to me to use one approach or the other, but most of these will work properly with either/or. If I had to choose one, I'd prefer the `onTabChanged` one because I like having single points of contact)

Reply:
Hmm.. since the context object for the whole select menu is handled in the main `SelectMenu` wrapper, that means we'd need to pass the `onTabChange` prop in to `SelectMenu` instead of `SelectMenu.Tabs` which feels kinda weird to me 🤔

Reply:
You could split it into multiple contexts; not every `SelectMenu` needs info about the logical `TabContext`